client/state/jetpack-connect/test/selectors.js

```diff
@@ -662,4 +664,55 @@ describe( 'selectors', () => {
 			expect( getAuthAttempts( state, 'sitetest.com' ) ).to.equals( 2 );
 		} );
 	} );
+
+	describe( '#getSiteIdFromQueryObject()', () => {
+		it( 'should return an integer', () => {
+			const state = {
+				jetpackConnect: {
+					jetpackConnectAuthorize: {
+						queryObject: {
+							client_id: '123'
+						}
+					}
+				}
+			};
+			expect( getSiteIdFromQueryObject( state ) ).to.equals( 123 );
+		} );
+
+		it( 'should return false if there is no query object', () => {
+			const state = {
+				jetpackConnect: {
+					jetpackConnectAuthorize: {}
+				}
+			};
+			expect( getSiteIdFromQueryObject( state ) ).to.equals( false );
+		} );
+
+		it( 'should return false if there is no client id', () => {
+			const state = {
+				jetpackConnect: {
+					jetpackConnectAuthorize: {
+						queryObject: {}
+					}
+				}
+			};
+			expect( getSiteIdFromQueryObject( state ) ).to.equals( false );
+		} );
+	} );
+
+	describe( '#hasNewlyConnectedSite()', () => {
+		it( 'should be undefined with an empty state', () => {
+			const state = {};
+			expect( hasNewlyConnectedSite( state ) ).to.be.undefined;
```

Comment:
The name of the selector (`has*`) implies that it returns a boolean value. How do you feel about returning a `false` instead of `undefined` in both of the cases where it now returns an `undefined`?